components/atom/button/src/index.scss

```diff
@@ -0,0 +1,180 @@
+$icon-m-button: $m-m !default;
+$p-button: $p-l !default;
+
+$sz-button-large: $p-base*6 !default;
+$fz-button-large: $fz-m !default;
+$icon-sz-button-large: $sz-icon-m !default;
+$icon-m-button-large: $m-m !default;
+$p-button-large: $p-l !default;
+
+// Set all color properties for text and icons
+@mixin button-foreground-color ($color) {
+  color: $color;
+
+
+  & svg path {
+    fill: $color;
+    stroke: $color;
+  }
+}
+// Sets active states when button is not disabled
+@mixin button-focused {
+  &:not(.sui-AtomButton--disabled) {
+    &.sui-AtomButton--focused,
+    &:active,
+    &:focus,
+    &:hover {
+      @content;
+    }
+  }
+}
+@mixin button-icon ($size, $margin) {
+  & .sui-AtomButton-leftIcon,
+  & .sui-AtomButton-rightIcon {
+    height: $size;
+    width: $size;
+  }
+
+  & .sui-AtomButton-leftIcon {
+    margin-right: $margin;
+  }
+
+  & .sui-AtomButton-rightIcon {
+    margin-left: $margin;
+  }
+}
+
+.sui-AtomButton {
+  @include reset-button;
+  @include button-icon($icon-sz-button, $icon-m-button);
+  border: 1px solid;
+  border-radius: $bdrs-m;
+  box-sizing: border-box;
+  font: {
+    family: $ff-sans-serif;
+    size: $fz-button;
+    weight: $fw-semi-bold;
+  }
+  height: $sz-button;
+  min-width: $sz-button;
+  outline: 0;
+  padding: 0 $p-button;
+  transition: background-color .3s ease, border-color .3s ease;
+  white-space: nowrap;
+
+  & + & {
+    margin-left: $m-m;
+  }
+
+  // Icons
+  &-leftIcon,
+  &-rightIcon {
+    display: inline-block;
+    transform: translateY(-2px);
+    vertical-align: middle;
+  }
+
+  // Types and colors
+  &--primary {
+    @include button-foreground-color($c-white);
+    @include button-focused {
+      background-color: $bgc-button-dark;
+      border-color: $bgc-button-dark;
+    }
+    background-color: $bgc-button;
+    border-color: $bgc-button;
+  }
+
+  &--secondary,
+  &--tertiary {
+    @include button-foreground-color($bgc-button);
+    @include button-focused {
+      background-color: $bgc-button-light;
+    }
+  }
+
+  &--tertiary {
+    border-color: transparent;
+  }
+
+  &--negative {
+    &.sui-AtomButton--primary {
+      @include button-foreground-color($bgc-button);
+      @include button-focused {
+        background-color: $bgc-button-light-contrast;
+        border-color: $bgc-button-light-contrast;
+      }
+      background-color: $c-white;
+      border-color: $c-white;
+    }
+
+    &.sui-AtomButton--secondary,
+    &.sui-AtomButton--tertiary {
+      @include button-foreground-color($c-white);
+      @include button-focused {
+        background-color: $bgc-button-dark;
+      }
+    }
+  }
+
+  &--accent {
+    @include button-foreground-color($c-white);
+    @include button-focused {
+      background-color: $c-accent-dark;
+      border-color: $c-accent-dark;
+    }
+    background-color: $c-accent;
+    border-color: $c-accent;
+  }
+
+  // Modifiers
+  &--disabled {
+    cursor: not-allowed;
+    opacity: .3;
```

Comment:
should we define a global opacity variable? even if we does not have an opacity specification, i would find it useful.

Reply:
I don't think so. This opacity value was specially defined for disable state of buttons.